client/src/test/java/org/eclipse/hono/client/CommandTest.java

```diff
@@ -212,4 +212,17 @@ public void testFromMessageFailsForNonMatchingReplyToAddress() {
                 CommandConstants.COMMAND_ENDPOINT, replyToId));
         assertFalse(Command.from(message, Constants.DEFAULT_TENANT, "4712").isValid());
     }
+
+    /**
+     * Verifies the return value of getInvalidCommandReason().
+     */
+    @Test
+    public void testGetInvalidCommandReason() {
+        final Message message = mock(Message.class);
+        when(message.getReplyTo()).thenReturn("invalid-reply-to");
+        final Command command = Command.from(message, Constants.DEFAULT_TENANT, "4712");
+        assertFalse(command.isValid());
+        assertThat(command.getInvalidCommandReason(),
+                is("subject not set, message/correlation id not set, replyTo not a command address: invalid-reply-to"));
```

Comment:
I do not really see the value of doing such literal error message comparisons. It only makes it harder to refactor later ...

Reply:
I've changed that:
In order to still have something to test in this method, I just try to match the different message field names there now.